### PR TITLE
Azure Automation Hybrid Worker resource commands added

### PR DESCRIPTION
**Pull request by krmanupa**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az automation hrwg create
az automation hrwg delete
az automation hrwg list
az automation hrwg show
az automation hrwg update

az automation hrwg hrw create
az automation hrwg hrw delete
az automation hrwg hrw list
az automation hrwg hrw move
az automation hrwg hrw show
az automation hrwg hrw update

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
no errors in the files added as part of this PR
- [X] Have you run `python scripts/ci/test_index.py -q` locally?
Yes, no errors were seen

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Automation

File: src/automation/azext_automation/azext_metadata.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
     "azext.isExperimental": true,
-    "azext.minCliCoreVersion": "2.13.0"
+    "azext.minCliCoreVersion": "2.39.0"
```
Comment (zhoxing-ms) [inline src/automation/azext_automation/azext_metadata.json:3]: Just out of curiosity, may I ask why are those new features only available to users who have installed the latest version of CLI?

**Comment by zhoxing-ms** [comment]
Please write the description of those new features into `HISTORY.rst`. In addition, if you want to release a new version of extension for this PR, please also modify the version defined in `setup.py`.

**Comment by zhoxing-ms** [comment]
@kairu-ms  Since this PR is generated by Code Gen v2, do you need to review it from the perspective of Code Gen expert?

File: src/automation/azext_automation/azext_metadata.json (hunk 1 above)
Comment (krmanupa) [reply src/automation/azext_automation/azext_metadata.json:3]: This has been auto updated when azdev tools were used. Do you want me to change it back to 2.13.0 ?

**Comment by zhoxing-ms** [comment]
@krmanupa Please address this CLI linter issue

File: src/automation/azext_automation/azext_metadata.json (hunk 1 above)
Comment (kairu-ms) [reply src/automation/azext_automation/azext_metadata.json:3]: 2.39.0 is required

**Comment by zhoxing-ms** [comment]
> az automation hybrid-runbook-worker-group hybrid-runbook-worker create
az automation hybrid-runbook-worker-group hybrid-runbook-worker delete
az automation hybrid-runbook-worker-group hybrid-runbook-worker list
az automation hybrid-runbook-worker-group hybrid-runbook-worker move
az automation hybrid-runbook-worker-group hybrid-runbook-worker show
az automation hybrid-runbook-worker-group hybrid-runbook-worker update

It seems that `hybrid-runbook-worker` has repeated the command naming. Could you optimize and simplify it?

File: src/automation/azext_automation/aaz/latest/automation/hybrid_runbook_worker_group/_create.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            required=True,
+        )
+
+        # define Arg Group "Properties"
+
+        _args_schema = cls._args_schema
+        _args_schema.credential = AAZObjectArg(
+            options=["--credential"],
+            arg_group="Properties",
+            help="Sets the credential of a worker group.",
```
Comment (zhoxing-ms) [inline src/automation/azext_automation/aaz/latest/automation/hybrid_runbook_worker_group/_create.py:66]: We need to use the first person voice in the help message
```suggestion
            help="Set the credential of a worker group.",
```

File: src/automation/azext_automation/aaz/latest/automation/hybrid_runbook_worker_group/_create.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        )
+
+        credential = cls._args_schema.credential
+        credential.name = AAZStrArg(
+            options=["name"],
+            help="Gets or sets the name of the credential.",
```
Comment (zhoxing-ms) [inline src/automation/azext_automation/aaz/latest/automation/hybrid_runbook_worker_group/_create.py:72]:
```suggestion
            help="Get or set the name of the credential.",
```

**Comment by zhoxing-ms** [comment]
> Please address this CLI linter issue

You can use `options_list` to declare a short alias for the parameter, or declare the check exclusion for this parameter in the `linter_exclusions.yml` file

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**